src/backend/aspen/workflows/import_locations/save.py

```diff
@@ -35,21 +39,42 @@ def save():
         )
         session.execute(gisaid_locations_insert)
 
-        # Make sure we have equivalent rows for region/country/division but with null locations
-        except_select = (
+        # Insert an entry with a null location for every distinct Region/Country/Division combination
+        # Doing a double-equals comparison is critical for the statement to compile to the intended SQL,
+        # so we tell flake8 to ignore rule E711
+        existing_null_location_select = (
```

Comment:
So generally, region > country > division > location. This select is choosing all the places where we are missing the most specific piece of info, but what happens for places where we're missing `location` _and_ `division` (or for that matter, `country` as well)? I can't tell from the SQL if it would return pre-existing NULL spots for more general places (eg, if there was a row of `Africa | Kenya | NULL |  NULL`): would those be selected with this statement?

Zooming out, we want to create entries for every place that are more general than the `location` level. Every distinct Region/Country/Division combo gets a generalized entry where its Location is NULL. Do we want to do that at even broader levels as well? Should there be entries for every distinct Region/Country combo where Division and Location are both NULL also? Or is the broadest scope we want to allow for the level of unknown Location? (Revisiting example above, would we want to have a general "Kenya" that's just Africa/Kenya/NULL/NULL? And similarly, would we even want a general "Africa" that's only Africa/NULL/NULL/NULL?) I honestly don't know what the Product needs on this are, so it's quite possible null-Location is broad enough, but I wanted to bring it up if it hasn't been discussed already.

Reply:
Yes, the broadest scope we want allow is division level. There are some entries like `Asia/Afghanistan/Afghanistan` but none where the division is null. We don't support that, and it's not in the gisaid metadata. And a sample just narrowed down to `Africa` is not useful at all!